### PR TITLE
Integrate Automattic about screen

### LINKED ISSUE #5435: Integrate Automattic About Screen Library
Goal of this issue is to replace the current About screen with the one implemented by @ravishanker - example : https://github.com/wordpress-mobile/WordPress-Android/pull/15529

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #5435

Merge Instructions:
1. Make sure https://github.com/Automattic/about-automattic-android/pull/29 is merged
2. Update hashcode
3. Remove "Do not merge" label
4. Review this PR
5. Merge this PR

### Description

This PR replaces the existing About screen with the unified About screen which is used across Automattic apps.

### Testing instructions

1. Tap on My Store
2. Tap on the gear icon in the top right corner
3. Scroll down and tap on "WooCommerce" row under the "About the app" section
4. Notice the new about screen is shown
5. Test clicking on all the rows and make sure they work as expected
6. Verify that "view shown" and "button click" events are being tracked

### Images/gif

| Before | After |
| --- | --- |
| Screenshot_1640776766 | 30-top-dark |
| | 30-bot-dark |
| Screenshot_1640776772 | 30-top-light |
| | 30-bot-light |

P.S. I considered adding unit tests but decided it's not worth the effort. About screen is static and we probably won't touch it for years.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

cc @ravishanker

### CODE REVIEW

File: WooCommerce/src/main/res/values/unified_about_themes.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="Theme.Woo.UnifiedAbout" parent="Theme.Woo.DayNight">
```
Comment (malinajirka) [inline WooCommerce/src/main/res/values/unified_about_themes.xml:3]: Modifying `colorSurface` directly feels hacky, but I couldn't think of a better solution. Any suggestions are welcome.

**Review by anitaa1990 — APPROVED** [review]
Looks good :shipit: Thank you so much for working on this @malinajirka <3. 

I'll refrain from merging till [this PR](https://github.com/Automattic/about-automattic-android/pull/29) is merged 👍

**Comment by AnirudhBhat** [comment]
@malinajirka Great job! Thanks for working on this. I noticed this weird behavior when I ran this on my device.

https://user-images.githubusercontent.com/1331230/147727119-ef1a2feb-3234-4b03-adc4-25dc7f575dd4.mp4

**Comment by malinajirka** [comment]
Thanks for the review @anitaa1990!

@AnirudhBhat Wow, that's interesting. What device were you using? Seems like I'm going to have some fun with Compose 😭 🥳 😃

**Comment by AnirudhBhat** [comment]
@malinajirka I ran this on Pixel 4a, Android 12. I noticed the same behavior when ran on an emulator which was running on Android 11 😄

**Comment by malinajirka** [comment]
I found the issue and also the reason why I missed it. The bug appears only on `jalapeno` builds, not on `wasabi` builds. My first thought was that it might be related to ProGuard/R8. However, it's disabled on debug builds not matter if it's jalapeno or wasabi. As it turned out it's related to the length of the app name -> when it doesn't fit one line this bug appears. I'll create a fix in the library. However, I believe we can merge this PR since the release build uses "WooCommerce" app name which is shorter than both "WooCommerce (Dev)" and "WooCommerce (Pre-Alpha)". Wdyt @AnirudhBhat ?

**Comment by AnirudhBhat** [comment]
@malinajirka Thanks for the investigation! Let's merge this :shipit:

**Review by AnirudhBhat — APPROVED** [review]

**Comment by malinajirka** [comment]
Just to close the loop,[ I filled a bug report in Android Compose Theme Adapter](https://github.com/material-components/material-components-android-compose-theme-adapter/issues/86). I'll update lineHeight from dp to sp in our codebase in a separate PR.

**Pull request merged**